modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -147,7 +151,23 @@ private ReturnDelegate createReturnDelegate() {
       return VoidReturnDelegate.INSTANCE;
     }
 
-    return StringUtils.isBlank(target) ? new ValueReturnDelegate(muleContext) : new TargetReturnDelegate(target, muleContext);
+    return !isTargetPresent() ? new ValueReturnDelegate(muleContext) : new TargetReturnDelegate(target, muleContext);
+  }
+
+  private boolean isTargetPresent() {
+    if (isBlank(target)) {
+      return false;
+    }
+
+    if (target.startsWith(FLOW_VARS)) {
+      throw new IllegalOperationException(format(INVALID_TARGET_MESSAGE, flowConstruct.getName(), operationModel.getName(),
+                                                 format("a flowVar with the '%s' prefix", FLOW_VARS)));
```

Comment:
this message is redundant. Read it to yourself and you'll notice that as a user you wouldn't like to receive that message